src/common/transformations/tests/common_optimizations/nop_elimination.cpp

```diff
@@ -1310,3 +1310,31 @@ TEST(SplitConcatElimination, no_sequence_found) {
     EXPECT_EQ(count_ops_of_type<ov::opset9::Split>(model), 1) << "SplitConcatElimination transformation has failed. "
                                                                  "The number of Split ops is not 1";
 }
+
+TEST(nop_elimination, gather_to_squeeze) {
+    auto generate_func = [](int64_t gather_axis) {
+        ov::Shape shape{3,3,4,4};
+        shape[gather_axis] = 1;
+        auto arg = std::make_shared<op::Parameter>(element::f32, shape);
+        auto indices = op::Constant::create(element::i64, Shape{}, vector<int64_t>{0});
+        auto axis = op::Constant::create(element::i64, Shape{}, vector<int64_t>{gather_axis});
+        auto gather = std::make_shared<op::v8::Gather>(arg, indices, axis);
+        return std::make_shared<Function>(NodeVector{gather}, ParameterVector{arg});
+    };
+
+    auto func_axis_0 = generate_func(0);
+    auto func_axis_1 = generate_func(1);
+    auto func_axis_2 = generate_func(2);
+    auto func_axis_3 = generate_func(3);
+    pass::Manager pass_manager;
+    pass_manager.register_pass<ov::pass::NopElimination>();
+    auto run_and_check = [&](std::shared_ptr<Function>& func) {
+        pass_manager.run_passes(func);
+        ASSERT_TRUE(count_ops_of_type<op::v8::Gather>(func) == 0);
+        ASSERT_TRUE(count_ops_of_type<op::v0::Squeeze>(func) == 1);
```

Comment:
Minor: I suggest using `EXPECT_EQ` instead of `ASSERT_TRUE` to be aligned with other instances in this file.